### PR TITLE
(PC-19293) feat(offer): add missing event param fromOfferId

### PR DESCRIPTION
**Pull request by YassinL**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-19293

## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](1)

## Screenshots

**delete** *if no UI change*

| Platform         | Before | After |
| :--------------- | :----: | :---: |
| iOS              |        |       |
| Android          |        |       |
| Phone - Chrome   |        |       |
| Tablet - Chrome  |        |       |
| Desktop - Chrome |        |       |
| Desktop - Safari |        |       |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

[PC-19293]: https://passculture.atlassian.net/browse/PC-19293?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/features/offer/pages/Offer/Offer.tsx [hunk 1]
```diff
@@ -40,10 +40,11 @@ export const Offer: FunctionComponent = () => {
   const { data: offer } = useOffer({ offerId })
   const similarOffers = useSimilarOffers(offerId, offer?.venue.coordinates)
   const hasSimilarOffers = similarOffers && similarOffers.length > 0
+  const fromOfferId = route.params?.fromOfferId
 
   const logSimilarOfferPlaylistVerticalScroll = useFunctionOnce(() => {
     if (hasSimilarOffers) {
-      return analytics.logSimilarOfferPlaylistVerticalScroll()
+      return analytics.logSimilarOfferPlaylistVerticalScroll(fromOfferId)
```
Comment (pierrecanthelou) [inline src/features/offer/pages/Offer/Offer.tsx:47]: La fonction accepte bien un paramètre null ?

File: src/libs/firebase/analytics/analytics.ts [hunk 2]
```diff
@@ -301,8 +301,10 @@ const logEventAnalytics = {
     analyticsProvider.logEvent(AnalyticsEvent.SIGN_UP_TOO_YOUNG, { age }),
   logSimilarOfferPlaylistHorizontalScroll: () =>
     analyticsProvider.logEvent(AnalyticsEvent.SIMILAR_OFFER_PLAYLIST_HORIZONTAL_SCROLL),
-  logSimilarOfferPlaylistVerticalScroll: () =>
-    analyticsProvider.logEvent(AnalyticsEvent.SIMILAR_OFFER_PLAYLIST_VERTICAL_SCROLL),
+  logSimilarOfferPlaylistVerticalScroll: (fromOfferId?: number) =>
+    analyticsProvider.logEvent(AnalyticsEvent.SIMILAR_OFFER_PLAYLIST_VERTICAL_SCROLL, {
```
Comment (pierrecanthelou) [inline src/libs/firebase/analytics/analytics.ts:305]: Idem ?
Ou bien faut-il faire un test avant ?

**Review by cedricls1 — APPROVED** [review]

File: src/features/offer/pages/Offer/Offer.tsx (hunk 1 above)
Comment (YassinL) [reply src/features/offer/pages/Offer/Offer.tsx:47]: Oui elle accepte bien un param null

File: src/libs/firebase/analytics/analytics.ts (hunk 2 above)
Comment (YassinL) [reply src/libs/firebase/analytics/analytics.ts:305]: Je peux rajouter un test pour check si j'ai bien `fromOfferId`

Comment (YassinL) [reply src/libs/firebase/analytics/analytics.ts:305]: [9ed35ee](https://github.com/pass-culture/pass-culture-app-native/pull/4151/commits/9ed35ee5d53f3aa4ccb3123495a323bb2d113437)

**Review by pierrecanthelou — APPROVED** [review]

**Pull request merged**